### PR TITLE
Lazy loading filter fix

### PR DESCRIPTION
**Pull request by amilosmanli**
Fixes filtering when lazy loading is enabled.

### CODE REVIEW

**Comment by amilosmanli** [comment]
cc: @benwinding would appreciate if you can take a look.

File: src/providers/lazy-loading/paramsToQuery.ts [hunk 1]
```diff
@@ -47,8 +47,14 @@ export function filtersToQuery(
...
 ): FireStoreQuery {
   const res = Object.entries(filters).reduce((acc,[fieldName, fieldValue]) => {
-    const opStr = fieldValue && Array.isArray(fieldValue) ? 'in' : '==';
-    return acc.where(fieldName, opStr, fieldValue);
+    if (Array.isArray(fieldValue)) {
+      return acc.where(fieldName, 'in', fieldValue);
+    } else {
+      return acc
+        .where(fieldName, '>=', fieldValue)
+        .where(fieldName, '<', fieldValue + 'z');
```
Comment (benwinding) [inline src/providers/lazy-loading/paramsToQuery.ts:55]: This is a fairly significant change, it changes the firebase query from an exact match `==`, to a greater or equal than match, is that what we want? Will that work for text and numbers?

Comment (amilosmanli) [reply src/providers/lazy-loading/paramsToQuery.ts:55]: Good question. Probably will not work as is with the numbers and need to add another case for number comparisons. 

This was the only way I could find that allows some sort of filtering for text with partial value

Comment (benwinding) [reply src/providers/lazy-loading/paramsToQuery.ts:59]: Thanks for the explanation, maybe we could handle the strict filtering case and partial filtering cases seperately.

> This was the only way I could find that allows some sort of filtering for text with partial value

I had no idea that the `>=` and `<` good match partial strings! 😮 

Will probably merge #255 first, but can merge this after

**Comment by amilosmanli** [comment]
Closing in favour of #273

**Pull request closed without merging**